app/scripts/modules/core/src/application/nav/NavCategory.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import { DataSourceNotifications } from '../../entityTag/notifications/DataSourceNotifications';
+import { Icon, useDataSource, useObservable } from '../../presentation';
+
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../../application';
+import { IEntityTags } from '../../domain';
+
+export interface INavCategoryProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavCategory: React.FC<INavCategoryProps> = ({ app, category, isActive }: INavCategoryProps) => {
+  const { alerts, badge, iconName, key, label, status$ } = category;
+
+  const { data: badgeData } = useDataSource(app.getDataSource(badge || key));
+  const runningCount = badge ? badgeData.length : 0;
+
+  const [tags, setTags] = React.useState<IEntityTags[]>(alerts || []);
+  useObservable(status$, () => {
+    setTags(category.alerts || []);
+  });
```

Comment:
Curious if we can use the more specific hook?
```suggestion
  const categoryData = useDataSource(category);
  const tags: IEntityTags[] = categoryData?.alerts || [];
```

Reply:
I did this because `alerts` isn't part of the data source's status object, so I did not think that `useDataSource` would subscribe correctly. I will play around though.

Reply:
oh I see!